pkgs/development/python-modules/opensfm/default.nix

```diff
@@ -57,6 +57,9 @@ buildPythonPackage rec {
   ];
 
   postPatch = ''
+    substituteInPlace opensfm/src/CMakeLists.txt \
+      --replace-fail "cmake_minimum_required(VERSION 3.0)" "cmake_minimum_required(VERSION 3.10)"
```

Comment:
I know it's objectionable, but I'd sort of rather see a regex instead of `3.0`, or at least a `.patch` so that the build only fails if the version cannot be relaxed cleanly

Reply:
We don't want to have a regex. If upstream fixes this line, we drop it again.

Reply:
It is impossible to support an arbitrary range of unknown versions and if the string no longer matches there are multiple causes for that. One of them is upstream fixed this but others could be upstream changed the code but not actually fixed the problem and silently skipping this in those cases is not great.

I assume you want to have an overlay with a totally different version. In that case you probably are better off copying parts of the derivation and adding an assert to be reminded when the version upstream changes.

Reply:
If you need another version of the package, why not add it to nixpkgs? That makes it easy for everyone to test for that, too and we do not step on each other shoes.